### DISCUSSION TITLE
[EssentialsChat] Improve Chat in console.

### DISCUSSION THREAD

**Discussion by Andre601** [EssentialsX/Essentials#3925] [Ideas and Feature Suggestions] [answered]
When you use custom hex colors for parts of the chat using EssentialsXChat is the last valid chat color getting used for it.

As an example: If you have the name displayed with hex color `FF0000` will the name be displayed as `§x<name>` in the console and the color will be black which - depending on the bg color of the console - may make the name/text invisible.

I assume this is the default Bukkit system for logging chat messages and it would perhaps be difficult to change it, but is that changeable?
If yes should it be considered as I feel like this could only make it more difficult to keep track of the chat properly.

Here's a screenshot of what I mean (Also executed `ess version` to share my setup).
![image](https://user-images.githubusercontent.com/11576465/104960392-69544000-59d4-11eb-95fa-493fd6298b1d.png)

**Comment by JRoy** [comment, marked as answer]
This is an upstream issue of spigot/paper itself. There's nothing we can do in this situation.

There have been multiple attempts to merge this into paper with various mixed reviews on the methodology that should be used 

https://github.com/PaperMC/Paper/pull/4221

**Reply by Andre601** [reply]
Already assumed that this might be the case. I'm glad that at least paper has a somewhat better logger implementation than Spigot.
I just hope there will be a future implementation of this, either through supporting more colors, or by just stripping away colors through like a config setting in paper.

Can you keep me updated if anything changes? Would be awesome.

**Reply by mdcfe** [reply]
Seems like Paper isn't going to PaperMC/Paper#4221 any time soon. Someone has released a [Paper plugin](https://papermc.io/forums/t/1-16-paperconsolehexpatch/6355) that may fix this. It's not a perfect solution as it reflects into the server to replace the console command sender, but might be of interest if you don't want to disable colours in your terminal entirely.

**Reply by pop4959** [reply]
Looks like this was addressed in https://github.com/PaperMC/Paper/commit/2a0ee4b65d63043564b0dfe001c7f9804a07b477 by @JRoy so this is no longer something we need to worry about (despite it not being an Essentials problem in the first place).
